static/app/components/replays/useReplaysCount.tsx

```diff
@@ -17,18 +17,35 @@ type Options = {
   transactionNames?: string | string[];
 };
 
+type CountState = Record<string, undefined | number>;
+
 function useReplaysCount({groupIds, transactionNames, organization, project}: Options) {
   const api = useApi();
   const location = useLocation();
 
-  const [replayCounts, setReplayCounts] = useState<Record<string, undefined | number>>(
-    {}
-  );
+  const [replayCounts, setReplayCounts] = useState<CountState>({});
+
+  const zeroCounts = useMemo(() => {
+    const ids = toArray(groupIds || []);
+    const names = toArray(transactionNames || []);
+    return ([] as string[])
+      .concat(ids)
+      .concat(names)
+      .reduce((record, key) => {
+        record[key] = 0;
+        return record;
+      }, {} as CountState);
```

Comment:
you can reduce the amount of typecasting
```suggestion
    return [...ids, ...names].reduce<CountState>((record, key) => {
      record[key] = 0;
      return record;
    }, {});
```